src/opm/output/eclipse/InteHEAD.cpp

```diff
@@ -633,4 +642,66 @@ Opm::RestartIO::InteHEAD::regionDimensions(const RegDims& rdim)
     this->data_[NMFIPR] = rdim.nmfipr;
 
     return *this;
-}
\ No newline at end of file
+}
+
+// =====================================================================
+// Free functions (calendar/time utilities)
+// =====================================================================
+
+namespace {
+    std::time_t advance(const std::time_t tp, const double sec)
+    {
+        using namespace std::chrono;
+
+        using TP      = time_point<system_clock>;
+        using DoubSec = duration<double, seconds::period>;
+
+        const auto t = system_clock::from_time_t(tp) +
+            duration_cast<TP::duration>(DoubSec(sec));
+
+        return system_clock::to_time_t(t);
+    }
+}
+
+std::time_t
+Opm::RestartIO::makeUTCTime(const std::tm& timePoint)
+{
+    auto       tp    =  timePoint; // Mutable copy.
+    const auto ltime =  std::mktime(&tp);
+    auto       tmval = *std::gmtime(&ltime); // Mutable.
```

Comment:
The call to gmtime() is not thread-safe, which can *in theory* cause problems as we actually use multiple threads for output. However, as the output writes are pipelined it is probably ok (if not, the restart file would be corrupted, which would be much more noticable than a wrong data somewhere...).

Reply:
>The call to `gmtime()` is not thread-safe,

Correct, and that's why we dereference the return value as soon as possible to get it into a local variable rather than relying on the internal static storage.  There will indeed be timing issues if multiple threads (from the same process) run this code.  If that's possible then we need to add some kind of synchronisation here.

Reply:
Ideally we would have the `date` library from C++20....